docs(cloud): surface Nextflow data lineage in Run details and Data Explorer (PLAT-1255) (#1361)
seqeralabs/docs: platform-cloud/docs/monitoring/run-details.mdx

Reviewer: ```suggestion
- **Name**: Process name and tag.
- **Status**: Exit code, task status, attempts.
- **Native ID**: Unique identifier assigned by the underlying execution executor to a specific job.
- **Command**: Task script, defined in the pipeline process.
- **Environment**: Environment variables supplied to the task.
- **Work directory**: Directory where the task was executed.
- **Inputs**: File inputs to the task and associated lineage data
- **Outputs**: File outputs from the task and associated lineage data
- **Upstream**: Links to related upstream tasks
- **Downstream**: Links to related downstream tasks
```

Reviewer: #### Metrics

- **Execution time**: Metrics for task submission, start, and completion time:

Reviewer: ```suggestion
  | **submitted** | Task submission timestamp. |
```

Reviewer: ```suggestion
  | **started** | Task execution timestamp. |
```

Reviewer: ```suggestion
  | **completed** | Task completion timestamp. |
```

Reviewer: ```suggestion
  | **total duration** | Time elapsed from task submission to completion, including scheduling time. |
```

Reviewer: ```suggestion
  | **script execution time** | Task script execution time. |
```

Reviewer: ```suggestion
- **Requested resources**: Metrics for the resources requested by the task:
```

Reviewer: ```suggestion
  | **container image** | Container image name used to execute the task. |
```

Reviewer: ```suggestion
  | **disk space** | Disk space requested for task executiuon. |
```

Reviewer: ```suggestion
  | **time limit** | Time requested for task execution. |
```

Reviewer: ```suggestion
| **cloudZone** | The cloud zone (region) where the task was executed. |
| **machineType** | The virtual machine type used for this task. |
```

Reviewer: ```suggestion
  | **estimated cost** | The estimated cost to compute this task. |
```

Reviewer: ```suggestion
- **Used resources**: Metrics for the actual resources used by the task:
```

- **Logs**: View and download the pipeline run's execution logs.
- **Metrics**: View resource [metrics](#wall-time) for the run.
- **Configuration**: View Nextflow configuration files and the resolved [configuration](#configuration) used for the run.
- **Inputs**: View pipeline parameters and input files used by the run, including their lineage records.
- **Outputs**: View files produced by the run, including reports and lineage records for every published file.
- **Containers**: View the details of containers used in the run, if any.
- **Run Info**: View details about the [run](#run-details), [infrastructure](#infrastructure-details), [executor](#executors-details), and [lineage](#lineage).

:::tip
Lineage-aware fields and tabs only display data when the run was executed with data lineage enabled. There are three ways to enable lineage:

> - [**Workspace settings → Lineage**][workspace-lineage-settings]. A workspace maintainer configures the cloud credentials, region, and (optionally) bucket name where lineage records are stored. Enable **Enable lineage by default** to make the launch-form lineage toggle default to on for every run launched in the workspace.

Author: ```suggestion
- [**Settings > Lineage**][workspace-lineage-settings]. A workspace maintainer configures the cloud credentials, region, and (optionally) bucket name where lineage records are stored. Select **Enable lineage by default** to make the launch form lineage toggle default to on for every run launched in the workspace.
```

- **Launch form toggle**. When launching a pipeline, toggle lineage on or off for the individual run.

> - **Nextflow configuration**. Set `lineage.enabled = true` in your pipeline's Nextflow config.

Reviewer 2: This won't work in current prod. Nextflow will produce lineage but platform won't capture it.

| **wchar** | Number of characters written to storage. |
| **vol_ctxt** | Number of voluntary context switches. |
| **inv_ctxt** | Number of involuntary context switches. |
| **lineage_id** | Lineage ID (LID) of the task's `TaskRun` record. Only populated when lineage tracking is enabled for the run. Select the LID to navigate to the lineage record. |

Use the search bar to filter tasks with substrings in the table columns such as **process**, **tag**, **hash**, and **status**. For example, if you enter `succeeded` in the **Search task** field, the table displays only tasks that succeeded.

#### Task details

![Task details](./_images/task-details.png)

> Select a task in the task table to open the **Task details** dialog. The dialog has the following tabs: **About**, **Execution log**, **Data Explorer**, **Container**, and — when lineage is enabled for the run — **Inputs** and **Outputs**.

Reviewer 3: ```suggestion
Select a task in the task table to open the **Task details** dialog. The dialog has the following tabs: **About**, **Execution log**, **Data Explorer**, and **Container**.  If lineage is enabled for the run, the dialog also includes **Inputs** and **Outputs** tabs.
```

> Select a task in the task table to open the **Task details** dialog. The dialog has the following tabs: **About**, **Execution log**, **Data Explorer**, and **Container**.  If lineage is enabled for the run, the dialog also includes **Inputs** and **Outputs** tabs.

Reviewer: This was missing the new **Metrics** tab and also the lineage info (wrongly placed below).

```suggestion
Select a task in the task table to open the **Task details** dialog. The dialog has the following tabs:

- **About**
- **Metrics**
- **Execution log**
- **Data Explorer**
- **Container** 

:::note
If lineage is enabled for the run, the **About** tab content includes **Inputs** and **Outputs** tabs.

The **Inputs** and **Ouputs** tabs show every input or output consumed by the task, including the name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the Lineage ID of the corresponding lineage record. Select a name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.
:::
```

| **wchar** | Number of characters written to storage. |
| **vol_ctxt** | Number of voluntary context switches. |
| **inv_ctxt** | Number of involuntary context switches. |
| **lineage_id** | Lineage ID (LID) of the task's `TaskRun` record. Populated only when lineage tracking is active for the run. Select the LID to navigate to the lineage record. |

Use the search bar to filter tasks with substrings in the table columns such as **process**, **tag**, **hash**, and **status**. For example, if you enter `succeeded` in the **Search task** field, the table displays only tasks that succeeded.

#### Task details

![Task details](./_images/task-details.png)

Select a task in the task table to open the **Task details** dialog. The dialog has the following tabs:

- **About**
- **Metrics**
- **Execution log**
- **Data Explorer**
- **Container**

:::note
If lineage is enabled for the run, the **About** tab content includes **Inputs** and **Outputs** tabs.

> The **Inputs** and **Ouputs** tabs show every input or output consumed by the task, including the name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the Lineage ID of the corresponding lineage record. Select a name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.

Author: ```suggestion
The **Inputs** and **Ouputs** tabs show every input or output consumed by the task, including the name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the lineage ID of the corresponding lineage record. Select a name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.
```

| **wchar** | Number of characters written to storage. |
| **vol_ctxt** | Number of voluntary context switches. |
| **inv_ctxt** | Number of involuntary context switches. |

> | **lineage_id** | Lineage ID (LID) of the task's `TaskRun` record. Populated only when lineage tracking is active for the run. Select the LID to navigate to the lineage record. |

Reviewer 2: Not in current prod.

Use the search bar to filter tasks with substrings in the table columns such as **process**, **tag**, **hash**, and **status**. For example, if you enter `succeeded` in the **Search task** field, the table displays only tasks that succeeded.

#### Task details

![Task details](./_images/task-details.png)

Select a task in the task table to open the **Task details** dialog. The dialog has the following tabs:

- **About**
- **Metrics**
- **Execution log**
- **Data Explorer**
- **Container**

:::note
If lineage is enabled for the run, the **About** tab content includes **Inputs** and **Outputs** tabs.

> The **Inputs** and **Outputs** tabs show every input or output consumed by the task, including the name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the lineage ID of the corresponding lineage record. Select a name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.

Reviewer 2: ```suggestion
The **Inputs** and **Outputs** tabs show every input or output consumed by the task, including the name and the lineage labels assigned to it. Select a name to open the file in [Data Explorer](../data/data-explorer).
```

| **Cached** | Indicates whether the container was previously built in an earlier request. |
| **Freeze** | Indicates whether the container was provisioned for persistent storage using Wave freeze mode. |

#### Inputs

The **Inputs** tab shows every input consumed by the task, including the input name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the Lineage ID of the corresponding lineage record. Select an input name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.

:::note
The Inputs and Outputs tabs are only displayed when [lineage tracking is enabled][nextflow-lineage-tutorial] for the run.
:::

#### Outputs

The **Outputs** tab shows every output produced by the task, including the output name (linked to the file in [Data Explorer](../data/data-explorer)), its lineage type, the destination path, lineage labels, and the Lineage ID of the `FileOutput` record.

> 

Reviewer: This is _per task_ and not it's own section. I have moved to the **Tasks** section above.

```suggestion
```

| **Cached** | Indicates whether the container was previously built in an earlier request. |
| **Freeze** | Indicates whether the container was provisioned for persistent storage using Wave freeze mode. |

#### Inputs

> The **Inputs** tab shows every input consumed by the task, including the input name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the Lineage ID of the corresponding lineage record. Select an input name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.

Author: ```suggestion
The **Inputs** tab shows every input consumed by the task, including the input name, its lineage type (`Collection` or `Path`), the source path, the lineage labels assigned to it, and the lineage ID of the corresponding lineage record. Select an input name to open the file in [Data Explorer](../data/data-explorer). Select a lineage ID or label to navigate to that lineage record.
```

</TabItem>
<TabItem value="Configuration" label="Configuration" default>

The **Configuration** tab contains information about the Nextflow configuration files and the Nextflow command used for the run.

:::note

> :::

Reviewer 3: ```suggestion
```

The **Parameters** window displays the pipeline parameters configured for the run, with options to view, copy, or download the parameters in Groovy, JSON, or YAML format.

#### Input files

The **Input files** table displays every dataset, file, and collection that was used as input for the run:

| Column | Description |
|--------|-------------|
| **Input Name** | Display name of the input. Select the name to open the file in [Data Explorer](../data/data-explorer) or in the corresponding [Dataset](../data/datasets). |
| **Type** | Lineage type, such as `Collection` or `Path`. |
| **File Path** | Full path to the input. The path is truncated; hover for the complete path. Select the path to open it in [Data Explorer](../data/data-explorer). |
| **Lineage ID** | Lineage ID (LID) of the input's lineage record. Only populated when [lineage tracking is enabled][nextflow-lineage-tutorial]. |
| **Lineage Labels** | Lineage labels assigned to the input. Each label is a clickable link that navigates to the lineage record for that label. |

If the run was not launched with any input files or datasets, the table is empty.

</TabItem>
<TabItem value="Outputs" label="Outputs" default>

> The **Outputs** tab lists every file the run published to its output directory. The tab has two sub-views:

Reviewer: ```suggestion
The **Outputs** tab links to every file the run published to its output directory. The tab has two sub-views:
```

> 

Reviewer: - **Pipeline outputs** - If the `outDir` parameter is used, or the run executed with the `-output-dir` option` all outputs can be viewed in Data Explorer.

- **Reports** — The named report files configured for the run, such as the MultiQC report or any reports declared in `tower.yml`. This is the same content that was previously displayed on the Reports tab.
- **All** — Every `FileOutput` record linked to the run. This view is populated when the run was executed with [lineage tracking enabled][nextflow-lineage-tutorial] and an `outputDir` (or `-output-dir`) configured.

> {/* TODO: design to supply ./_images/outputs.png screenshot of the Outputs tab. Re-add the image reference once the asset is committed: ![Outputs](./_images/outputs.png) */}

Reviewer: Wrong. Not in this milestone.

```suggestion
```

The **Reports** sub-tab contains a table with the names, details, and paths to all [reports](../reports/overview) generated by the run, if any were configured. Select a report to open a Data Explorer file preview of the report, with options to open the report in a new tab or download it.

#### All

The **All** sub-tab lists every file output from the run that has a corresponding `FileOutput` lineage record:

| Column | Description |
|--------|-------------|
| **Output Name** | Display name of the output. Select the name to open the file in [Data Explorer](../data/data-explorer). |
| **Type** | Lineage type, such as `Collection` or `Path`. |
| **Process** | The Nextflow process that produced the output. |
| **File Path** | Full path to the output. The path is truncated; hover for the complete path. Select the path to open it in [Data Explorer](../data/data-explorer). |
| **Lineage ID** | Lineage ID (LID) of the `FileOutput` record. |
| **Lineage Labels** | Lineage labels assigned by the `label` directive. Each label is a clickable link that navigates to the lineage record for that label. See the [Nextflow `label` directive][nextflow-label-directive] for assignment details. |

> If the run was not launched with lineage enabled, the **All** sub-tab is empty and a note explains how to enable lineage.

Reviewer: Wrong. Not in this milestone.

```suggestion
```

The **Command** window displays the Nextflow command used for the run.

</TabItem>
<TabItem value="Inputs" label="Inputs" default>

The **Inputs** tab consolidates the pipeline parameters and the input files used by the run.

#### Parameters

![Parameters](./_images/parameters.png)

The **Parameters** window displays the pipeline parameters configured for the run, with options to view, copy, or download the parameters in Groovy, JSON, or YAML format.

> #### Input files

Reviewer 4: This is not "Input files" yet but still just "Datasets" as before basically - so the "Inputs" tab is just `Parameters` + `Datasets` (but no lineage information there)

</TabItem>
<TabItem value="Run Info" label="Run Info" default>

> The **Run Info** tab contains at-a-glance details about the run, infrastructure, executor, and — when lineage tracking is enabled — lineage. Hover over the information icon next to a card's name to view a value description. Select the icons next to any run detail values to copy them.

Reviewer 3: ```suggestion
The **Run Info** tab contains at-a-glance details about the run, infrastructure, and executor. When lineage tracking is enabled, it also displays lineage information. Hover over the information icon next to a card's name to view a value description. Select the icons next to any run detail values to copy them.
```

| **Pipeline ID** | `platform.pipeline.id` | Seqera Platform pipeline ID. |
| **Pipeline Name** | `platform.pipeline.name` | Seqera Platform pipeline name. |
| **Pipeline Revision** | `platform.pipeline.revision` | Pipeline revision used for the run. |
| **Pipeline Commit ID** | `platform.pipeline.commitId` | Pipeline Git commit ID used for the run. |

#### Empty and error states

| Condition | Behavior |
|-----------|----------|
| Lineage was not enabled for this run | The Lineage view is hidden. A note reads: *"Lineage tracking was not enabled for this run. Add `lineage.enabled = true` to your Nextflow config to capture lineage data, or toggle lineage on in the launch form."* To make lineage the default for all runs in this workspace, configure [**Workspace settings → Lineage**][workspace-lineage-settings] and enable **Enable lineage by default**. |
| Lineage was enabled but no records were found | The Lineage view displays: *"No workflow lineage records found for this run."* The Inputs, Outputs, and Tasks tabs each display the equivalent message for their record type. |
| The lineage indexer is unavailable | The Lineage view displays an error state with a retry action. |

> 

Reviewer: Repeat from the last review: this is not included in this release.

| **Compute Env Platform** | `platform.computeEnv.platform` | The compute environment platform (for example, `aws-cloud`). |
| **Pipeline ID** | `platform.pipeline.id` | Seqera Platform pipeline ID. |
| **Pipeline Name** | `platform.pipeline.name` | Seqera Platform pipeline name. |
| **Pipeline Revision** | `platform.pipeline.revision` | Pipeline revision used for the run. |
| **Pipeline Commit ID** | `platform.pipeline.commitId` | Pipeline Git commit ID used for the run. |

#### Empty and error states

| Condition | Behavior |
|-----------|----------|
| Lineage was not enabled for this run | The Lineage view is hidden. A note reads: *"Lineage tracking was not enabled for this run. Add `lineage.enabled = true` to your Nextflow config to capture lineage data, or toggle lineage on in the launch form."* |
| Lineage was enabled but no records were found | The Lineage view displays: *"No workflow lineage records found for this run."* The Inputs, Outputs, and Tasks tabs each display the equivalent message for their record type. |
| The lineage indexer is unavailable | The Lineage view displays an error state with a retry action. |

> 

Reviewer: We're not exposing this in platform in this milestone.

```suggestion
```
